nodejs/eks/cluster.ts

```diff
@@ -1258,14 +1324,18 @@ export class Cluster extends pulumi.ComponentResource {
      * https://docs.aws.amazon.com/eks/latest/userguide/worker.html
      */
     createNodeGroup(name: string, args: ClusterNodeGroupOptions): NodeGroup {
+        const awsProvider = this.core.awsProvider ? {aws: this.core.awsProvider} : undefined;
         return new NodeGroup(name, {
             ...args,
             cluster: this.core,
             nodeSecurityGroup: this.core.nodeGroupOptions.nodeSecurityGroup,
             clusterIngressRule: this.core.nodeGroupOptions.clusterIngressRule,
         }, {
             parent: this,
-            providers: { kubernetes: this.provider },
+            providers: {
+                ...awsProvider,
```

Comment:
Does this work if `awsProvider` is `undefined`? I think the spread operator only applies to object types.

Reply:
Yes - https://www.bennadel.com/blog/3590-you-can-safely-apply-object-spread-to-null-and-undefined-values-in-typescript-3-2-4.htm